modules/behavioural_qc/php/conflicts.class.inc

```diff
@@ -0,0 +1,151 @@
+<?php declare(strict_types=1);
+
+namespace LORIS\behavioural_qc;
+
+use LORIS\behavioural_qc\Provisioners\ConflictsProvisioner;
+use LORIS\Data\Filters\HasAnyPermissionOrUserSiteMatch;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Middleware\ETagCalculator;
+
+/**
+ * The Conflicts Class.
+ *
+ * GET request method for endpoint in behavioural_qc module.
+ * Used for populating the Data Conflicts table.
+ *
+ * @category Behavioural
+ * @package  Loris
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Conflicts extends \NDB_Page implements ETagCalculator
+{
+    /**
+     * A cache of the endpoint results, so that it doesn't need to be
+     * recalculated for the ETag and handler.
+     */
+    private $_cache;
+
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET request.
```

Comment:
You're never calling the parent class, this is losing the default behavior of NDB_Page (such as checking permissions). You're also never doing any validating of the url. `/conflicts` `/conflicts/abc` and `/conflicts/abcasdf/23?234` will all return the same results

Reply:
@driusan sorry I'm a little confused on what you mean for your first sentence. The _hasAccess does get called? I'm not sure what validation I didn't port over because this PR was to reactify the module.

Reply:
My mistake, it's process that calls hasAccess, not handle.